### ISSUE TITLE
Updating problems

### ISSUE THREAD

**Issue by efremsternbach** [microsoft/vscode#94794]
Issue Type: <b>Bug</b>

I've been having trouble when updating Visual Studio Code. Whenever I try to update over the last few months there are files in the resource directory that are locked and block the installation. The installation is not rolled back but is corrupted. This happens whether I use either the system or user installer. I can't delete the files even with Administrator permissions and I have to do a lot of manual work including rebooting the computer to remove the corrupted installation.

If there are files that continue to be used by some process the process has no recognizable name.  I tried using process explorer to identify what is locking it but it pointed to explorer.exe. On the latest upgrade attempt the following files are locked :

\resources\app\out\vs\platform\files\node\watcher\win32\CodeHelper.exe
\resources\app\node_modules.asar.unpacked\vscode-ripgrep\bin\rg.exe

This is in my personal AppData directory, not a system directory. For the last update (1.44) I actually had to reboot in safe mode before I could remove the leftover files.

I have Windows Defender and Malwarebytes running on this computer. I also have been using Docker Desktop to debug AWS Lambda functions. Not sure why any of those would cause the files to be locked but just in case.

VS Code version: Code 1.44.0 (2aae1f26c72891c399f860409176fe435a154b13, 2020-04-07T23:31:18.860Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz (8 x 3408)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.88GB (18.86GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (39)</summary>

Extension|Author (truncated)|Version
---|---|---
aws-toolkit-vscode|ama|1.8.0
pgformatter|bra|1.14.0
npm-intellisense|chr|1.3.0
path-intellisense|chr|1.4.2
vscode-markdownlint|Dav|0.34.0
jshint|dba|0.10.21
vscode-eslint|dba|2.1.2
xml|Dot|2.5.0
gitlens|eam|10.2.1
prettier-vscode|esb|4.1.1
vscode-firefox-debug|fir|2.7.1
json2ts|Gre|0.0.6
vscode-guid|hea|1.4.19
beautify|Hoo|1.5.0
ts-extension-pack|loi|0.2.0
rainbow-csv|mec|1.6.0
python|ms-|2020.3.71659
remote-containers|ms-|0.112.0
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.2
vscode-remote-extensionpack|ms-|0.20.0
node-debug2|ms-|1.42.1
powershell|ms-|2020.3.0
vscode-typescript-tslint-plugin|ms-|1.2.3
debugger-for-chrome|msj|4.12.6
debugger-for-edge|msj|1.0.14
sqltools|mtx|0.21.6
vscode-json-editor|nic|0.2.3
vscode-versionlens|pfl|0.25.0
tsimporter|pmn|2.0.1
typescript-hero|rbb|3.0.0
rust|rus|0.7.0
yo|sam|0.9.3
markdown-preview-enhanced|shd|0.5.2
move-ts|str|1.12.0
node-pack|Swe|0.1.10
html-preview-vscode|tht|0.2.5
vscodeintellicode|Vis|1.2.6

</details>

**Comment by joaomoreno** [comment]
Can you reproduce if you disable Windows background updates? Set `update.enableWindowsBackgroundUpdates` to `false`.

**Comment by efremsternbach** [comment]
I'll look at this tonight. Last time I updated it took me over an hour to clean the wreckage. I can't afford to try this during the work day.

**Comment by vfrazao-ns1** [comment]
I've been seeing this same issue as well. It seems to be the same usual files where somehow the permissions get corrupted to the point where I can't remove the files even with admin privileges.

Right now the problem files appear to be:
```
\AppData\Local\Programs\Microsoft VS Code\resources\app\node_modules.asar.unpacked\vscode-ripgrep\bin
```  

and

```
'\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\platform\files\node\watcher\win32\CodeHelper.exe'
```

Same as @efremsternbach 

This happened the last time that I exited VS Code and the automatic updater tried to install the newest version so I can't grab any info or config from VS Code to provide here.

**Comment by vfrazao-ns1** [comment]
Seems like this may be related (the same?) as: `https://github.com/microsoft/vscode/issues/47841`  and `https://github.com/microsoft/vscode/issues/90721`

**Comment by efremsternbach** [comment]
I just tried unchecking the setting for background updates (go to Settings and filter for 'update') and I kicked off an update to 1.44.1
This time everything worked without problems  I don't know if this represents a fix for the problem in the 1.44.1 installer or if unchecking that setting did the job. If you've had the same problem I'd be glad to hear your results

**Comment by joaomoreno** [comment]
I'll close this as duplicate of #47841 with a recommendation to disable background updates.
